homeassistant/components/vallox/config_flow.py

```diff
@@ -0,0 +1,125 @@
+"""Config flow for the Vallox integration."""
+from __future__ import annotations
+
+from contextlib import suppress
+import ipaddress
+import logging
+import re
+from typing import Any
+
+from vallox_websocket_api import Vallox
+from vallox_websocket_api.exceptions import ValloxApiException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DEFAULT_NAME, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): str,
+    }
+)
+
+
+def host_valid(host: str) -> bool:
+    """Return True if hostname or IP address is valid."""
+    with suppress(ValueError):
+        if ipaddress.ip_address(host).version in [4, 6]:
+            return True
+
+    disallowed = re.compile(r"[^a-zA-Z\d\-]")
+    return all(x and not disallowed.search(x) for x in host.split("."))
+
+
+async def validate_host(hass: HomeAssistant, host: str, name: str) -> dict[str, Any]:
+    """Validate that the user input allows us to connect."""
+
+    if not host_valid(host):
+        raise InvalidHost(f"Invalid host: {host}")
+
+    client = Vallox(host)
+
+    try:
+        info = await client.get_info()
+    except (OSError, ValloxApiException) as err:
+        raise CannotConnect("Failed to connect") from err
+
+    return {"title": name, "model": info["model"]}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for the Vallox integration."""
+
+    VERSION = 1
+
+    async def async_step_import(self, data: dict[str, Any]) -> FlowResult:
+        """Handle import from YAML."""
+        name = data.get(CONF_NAME, DEFAULT_NAME)
+        host = data[CONF_HOST]
+
+        self._async_abort_entries_match({CONF_HOST: host})
+
+        reason = None
+        try:
+            info = await validate_host(self.hass, host, name)
+        except InvalidHost:
+            _LOGGER.exception("An invalid host is configured for Vallox")
+            reason = "invalid_host"
+        except CannotConnect:
+            _LOGGER.exception("Cannot connect to Vallox")
+            reason = "cannot_connect"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            reason = "unknown"
+        else:
+            return self.async_create_entry(title=info["title"], data=data)
```

Comment:
The title here is the `name` you passed to `validate_host`.  Instead of passing the `name`, just return a `bool` from `validate_host` since the model isn't used anyways

Reply:
Or even better, don't return anything and just rely on an exception being thrown? 🤔

Reply:
You could make it simplier and drop `CannotConnect` and make `(OSError, ValloxApiException)` a constant to replace it.

Reply:
True that. Fixed the exception part, waiting with making any changes to what the method returns on the happy path until we have settled that discussion.

Reply:
Changed the `validate_host` method to not return anything if successful as we won't use the model as the default name.